syntax = "proto3";

service cliente_name {
    rpc enviar_metadata (informacion_archivo ) returns (metadata);
}

message informacion_archivo  {
    string nombre_archivo = 1;
    int32 numero_bloques = 2;
}

message metadata {
    repeated string message = 1;
}

service cliente_data {
    rpc enviar_bloques (informacion_bloque) returns (confirmacion);
}

message informacion_bloque {
    bytes bloque = 1;
    string id = 2;
    string nombre = 3;
}

message confirmacion {
    string message = 1;
}